packages/lib/src/classes/ApplicationCommandHandler.ts

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable guard-for-in */
+import { Collection, ApplicationCommandData, CommandInteraction } from "discord.js"
+import { BetterClient, ApplicationCommand, _BaseHandler, _BaseComponent, HandlerType, LogLevel } from "../index.js"
+import { generateErrorMessage } from "@buape/functions"
+
+export default class ApplicationCommandHandler extends _BaseHandler {
+	// Key is `${userID}-${commandName}`.
+	private cooldowns: Collection<string, number>
+
+	constructor(client: BetterClient) {
+		super(HandlerType.ApplicationCommand, client)
+		this.cooldowns = new Collection()
+	}
+
+	public override postLoad() {
+		return setTimeout(async () => {
+			this.client.application?.commands.set(
+				this.client.applicationCommands.map((command) => {
+					const data = this.getDiscordCommandData(command)
+					return data
+				})
+			)
+			await Promise.all(
+				this.client.guilds.cache.map((guild) =>
+					guild.commands.set([]).catch((error) => {
+						if (error.code === 50001) {
+							this.client.log(
+								`I encountered DiscordAPIError: Missing Access in ${guild.name} [${guild.id}] when trying to set slash commands!`,
+								LogLevel.ERROR
+							)
+						} else {
+							this.client.log(`${error}`, LogLevel.ERROR)
+						}
```

Comment:
This `if` literally logs the exact same thing except it says that it happened while registering slash commands. Can be removed and just keep the `this.client.log(error)` and if the logger supports a `message` property say in there that it was while registering slash commands. `error` will have the error code and error name because of discord.js; also why `this.client.log(`${error}`, LogLevel.ERROR)` not `this.client.error(error)`?